### ISSUE TITLE
Question on Unbalanced Quotes 

### ISSUE THREAD

**Issue by mide** [adrienverge/yamllint#397]
Hi there! Long time user and fan of this project - It keeps everything nice and tidy, so thanks for the work!

## Background
I had a question on unbalanced quotes, and what the expectations are around them. Specifically, when I attempt to lint this block
```yaml
---
BadStringMissingEnd: "Hello World
```
the linter errors (as I expect), but when I run against this block
```yaml
---
BadStringMissingStart: Hello World"
```
it passes, to my surprise. 

## Questions

These are the questions I have, hopefully I'm looking in the right place:
1. Is this the expected behavior according the the Yaml specification? 
2. Is this the expected behavior of this tool? 
3. Is there something I can do to catch the `BadStringMissingStart` error case with configuration? _(Outside of `rules:
  quoted-strings: required: true` which I'd like to avoid if possible)_

## Additional Context / Examples

This is the script I used to produce the examples, along with the output. 

```sh
yamllint --version
> yamllint 1.26.1

# Expected result
echo -e "---\nGoodStringFullQuote: \"Hello World\"" > GoodStringFullQuote.yaml
yamllint -s GoodStringFullQuote.yaml
echo $?
> 0

# Expected result
echo -e "---\nGoodStringNoQuote: Hello World" > GoodStringNoQuote.yaml
yamllint -s GoodStringNoQuote.yaml
echo $?
> 0

# Expected result
echo -e "---\nBadStringMissingEnd: \"Hello World" > BadStringMissingEnd.yaml
yamllint -s BadStringMissingEnd.yaml
> BadStringMissingEnd.yaml
>  3:1       error    syntax error: found unexpected end of stream (syntax)
echo $?
> 1

# Unexpected result
echo -e "---\nBadStringMissingStart: Hello World\"" > BadStringMissingStart.yaml
yamllint -s BadStringMissingStart.yaml
echo $?
> 0
```

Thanks!

**Comment by adrienverge** [comment]
Hello, thanks for the interesting question :+1:

> 1. Is this the expected behavior according the the Yaml specification?

According to the [YAML 1.2 specification about Plain Style](https://yaml.org/spec/1.2/spec.html#id2788859), it seems to be valid, yes.

Please note that the `"` character is part of the string, hence your example is equivalent to:

```yaml
BadStringMissingStart: "Hello World\""
```

Also, if you'd wanted to escape `"` while keeping the *plain string* notation (no surrounding quotes), you could not, because `\` is treated "raw" too. For example:
```yaml
BadStringMissingStart: Hello World\"
```
has the same value as:
```yaml
BadStringMissingStart: "Hello World\\\""
```

> 2. Is this the expected behavior of this tool?

If a notation is valid, there is no reason for reporting it, in my opinion. They are other reserved characters with special meanings in YAML (e.g. `-`, `:`, `#`...) that *change the meaning of the YAML notation when they start a string* -- however we don't report them when they're in the 2nd or *n*th position in the string.

> 3. Is there something I can do to catch the `BadStringMissingStart` error case with configuration? _(Outside of `rules:   quoted-strings: required: true` which I'd like to avoid if possible)_

I only see the `quoted-strings: {required: true}`... (sorry :))
Another approach would be to write a custom script that checks the content of YAML *plain* strings, and report when they contain quotes.

I have a question, for curiosity: why is it a problem to allow `"` chars inside *plain* strings? Is it only for the visual appearance (which I can understand), or is there something else?

**Comment by mide** [comment]
Thanks for the quick response. 
> If a notation is valid, there is no reason for reporting it, in my opinion.
> I have a question, for curiosity: why is it a problem to allow `"` chars inside plain strings? Is it only for the visual appearance (which I can understand), or is there something else?

The Yaml files we're writing are being consumed by a 3rd party (Specifically [AWS CloudFormation](https://aws.amazon.com/cloudformation/)) and the trailing quote can cause unexpected errors there. We were hoping to catch the errors in an earlier stage to speed up detection and resolution. 

Thanks for the help, we'll have to write some custom tooling!
